### ISSUE TITLE
[BUG] - Existing certificate secret name is not being picked up.

### ISSUE THREAD

**Issue by abilal-mss** [nebari-dev/nebari#1581]
### Describe the bug

When using a existing certificate type existing and providing secret name in the nebari config file, the secret name isn't being passed into the manifested terraform files. 

### Expected behavior

We should be to utilize an existing secret for our tls certificate. 

### OS and architecture in which you are running Nebari

AWS, EKS

### How to Reproduce the problem?

When adding this code block in our nebari config file and then running this command `nebari deploy -c nebari-config.yaml --dns-provider cloudflare --dns-auto-provision`
```
certificate:
  type: existing
  secret_name: my-tls-certificate-secret
```

### Command output

```bash session
[terraform]:   # module.kubernetes-ingress.kubernetes_manifest.tlsstore_default[0] will be updated in-place
[terraform]:   ~ resource "kubernetes_manifest" "tlsstore_default" {
[terraform]:       ~ object   = {
[terraform]:           ~ spec       = {
[terraform]:               ~ defaultCertificate = {
[terraform]:                   ~ secretName = "my-tls-certificate-secret" -> ""
[terraform]:                 }
[terraform]:             }
[terraform]:             # (3 unchanged elements hidden)
[terraform]:         }
[terraform]:         # (1 unchanged attribute hidden)
[terraform]:     }
```

### Compute environment

AWS

### Anything else?

Researching past changes, appears issue might originated from this change https://github.com/nebari-dev/nebari/pull/1421/files#diff-a77d45450ea0a454e8b346914975631f93dc83a04d3886ec9e99977703a93164L191.

**Comment by iameskild** [comment]
Hi @abilal-mss, thank you for bringing this to our attention! Let us test try to recreate this issue and get back to you. In the meantime, would you mind describing how you generated your cert and created your secret. Thanks again :)

**Comment by sblair-metrostar** [comment]
I work with @abilal-mss, we encountered a couple of issues this week that ended with this certificate bug. Here's a brief rundown:

- Over the weekend, one of our users reported being unable to start single user environments or dask workers
- Spent several hours between Monday and Tuesday troubleshooting as the problem got worse, eventually all pods were found to be nearly impossible to schedule due to timeouts mounting volumes of basically any type. The issue initially sounded a lot like [this](https://github.com/nebari-dev/nebari/issues/1163) when it was just the jupyter/dask problems, but escalated quickly
- In the course of troubleshooting, I replaced several of the worker nodes and forced redeployments of virtually everything on the cluster multiple times
- Eventually things started working again, I assume due to all the fresh worker nodes but still no good explanation for it. I'm guessing it'll happen again once the cluster has been running in this state a while.
- Discovered once everything was back up that our traefik ingress let's encrypt certificate was failing and the site was presenting the self signed default certificate
- Figured out pretty quickly that this was due to having hit the let's encrypt limit and been forced into a backoff period that would take a day or two to get our cert back. I haven't dug into it too far but it looks like the automated let's encrypt certificate managed by traefik isn't actually persisted anywhere? All the restarts while dealing with the volume issue apparently caused several certificates to be requested which pushed us over the limit.

**In response to your actual question**
- To get the cert back up, I installed [cert-manager](https://cert-manager.io/) and set it up to issue a cert from zerossl and save to a secret, in this example we'll call it `my-tls-certificate-secret`. We'd done this on several QHub clusters previously so were confident it would work. Also since the issued cert is saved to a secret, we shouldn't need to request so many next time an issue like this occurs.
- We then reconfigured the nebari-config.yaml with the snippet @abilal-mss included above and ran the update. No change to the TLSStore was detected, leaving the certificate name empty, but it did update traefik to remove the letsencrypt settings and replace with the default values used by the `type: existing`.
- In the interests of time, I manually edited the TLSStore object with the new secret name to get things working, observed the valid zerossl issued cert on the site and everything was otherwise working.
- Subsequent attempts to run the nebari deploy command, unfortunately, result in terraform thinking it needs to empty out the secret name again before throwing an error that the object is out of sync with the state and failing to apply. All changes we've made since for any reason have required us to delete the TLSStore from the cluster first, allow nebari deploy and terraform to apply the blank TLSStore object, then manually go back in and edit the TLSStore with the correct secret name to fix the cert.

Sorry for the long winded response, but I thought the context would be valuable and we are likely to be reporting the volume mounting issues that precipitated all this eventually once I can better articulate the problem and describe the symptoms.

**Comment by viniciusdc** [comment]
Hi @sblair-metrostar thanks for raising these points. Reggarding the issue with the mount volumes, were you able to identify the original volume that presented the error?

Sometimes, you can try to restart the machine assigned to each node group, this will re-trigger all resources in the group to be re-generated (including errored volume claims) -- On gcp the option is called restart cluster node, but it would require some googling to identify what it looks like in the other providers.

Regarding the TLS cert error, I know this is a silly question but after switching to the existing mode, have you added the secret to the Kubernetes namespace? Something like:
```
kubectl create secret tls <tls-secret-alias> --cert=path/to/cert/file --key=path/to/key/file -n <namespace>
```
in theory, that should have worked for almost all nebari-related resources.

**Comment by sblair-metrostar** [comment]
The secret is there, everything works so long as the TLSStore resource is updated to look for it. It just looks like the `secret_name` parameter from nebari_config.yaml isn't being passed through to Terraform so it's always empty after an apply.

``` yaml
apiVersion: traefik.containo.us/v1alpha1
kind: TLSStore
metadata:
  creationTimestamp: "2022-12-06T15:34:48Z"
  generation: 2
  name: default
  namespace: onyx
  resourceVersion: "39052"
  uid: 5db2cdbb-a6a8-49b4-a116-c1a7eaa13317
spec:
  defaultCertificate:
    secretName: metrostar-certificate
```

That `secretName` is an empty string after Terraform runs, even though it is an input variable that based on everything I can find in the [code](https://github.com/nebari-dev/nebari/blob/784ab8147ef01a6eaa443c047dd9666e3722c49a/nebari/template/stages/04-kubernetes-ingress/modules/kubernetes/ingress/main.tf#L346) and docs as well as known working behavior in prior versions of QHub would work as expected. It simply appears to be getting dropped somewhere in the middle at this point, my best guess is the change made [here](https://github.com/nebari-dev/nebari/pull/1421/files#diff-a77d45450ea0a454e8b346914975631f93dc83a04d3886ec9e99977703a93164L191) which was the last place I found in the history where that variable was explicitly populated. @abilal-mss and I observed the same behavior on a fresh nebari install yesterday so I don't think it has anything necessarily to do with the environment that experienced all the volume issues, nor can I guess how those could be related but crazier things have happened.

Also as a semi-related but way less important potential bug with that TLSStore, there's a [condition](https://github.com/nebari-dev/nebari/blob/784ab8147ef01a6eaa443c047dd9666e3722c49a/nebari/template/stages/04-kubernetes-ingress/modules/kubernetes/ingress/main.tf#L336) on its creation that suggests it's not meant to be there at all when the `certificate-secret-name` variable is null. But that variable is being defaulted to an empty string [further upstream](https://github.com/nebari-dev/nebari/blob/784ab8147ef01a6eaa443c047dd9666e3722c49a/nebari/template/stages/04-kubernetes-ingress/variables.tf#L42) so I'm not sure it ever is null, or at least I haven't seen a case when that condition is evaluating to false in my testing. Not sure it matters but just fyi.

Regarding the volume mounting issue, for the first day or so it was only the conda-store nfs volume failing to mount which was preventing Jupyter single user pods as well as dask workers for any existing user environments from spinning up. It wasn't 100% failing at first, started off with intermittent failures then gradually increased in consistency throughout the day to the point that no new pods would schedule. Besides the obvious investigations of pod logs for everything that seemed relevant, checking the node available storage, confirming the conda nfs service was running and accessible, redeploying pretty much everything in the nebari namespace, wasn't having much luck pinning down a cause.

When it reached the point that everything was failing I was mostly in fire-fighting mode and just had to get things back up, which is when I started terminating nodes to get fresh instances. Whether that fixed it or a reboot of the node would have sufficed I can't say, but I'm not sure there's really much difference at the end of the day.
